pkg/consensus/payload.go

```diff
@@ -195,18 +196,34 @@ func (p *Payload) Sign(key *privateKey) error {
 }
 
 // Verify verifies payload using provided Witness.
-func (p *Payload) Verify() bool {
-	h := sha256.Sum256(p.MarshalUnsigned())
+func (p *Payload) Verify(scriptHash util.Uint160) bool {
+	verification := p.Witness.VerificationScript
+	if len(verification) == 0 {
+		bb := new(bytes.Buffer)
+		err := vm.EmitAppCall(bb, scriptHash, false)
+		if err != nil {
+			return false
+		}
+		verification = bb.Bytes()
+	} else if h := p.Witness.ScriptHash(); scriptHash != h {
+		return false
+	}
+
 	v := vm.New()
+	h := sha256.Sum256(p.MarshalUnsigned())
+
 	v.SetCheckedHash(h[:])
-	v.Load(append(p.Witness.InvocationScript, p.Witness.VerificationScript...))
-	if err := v.Run(); err != nil || v.Estack().Len() == 0 {
+	v.LoadScript(verification)
+	v.LoadScript(p.Witness.InvocationScript)
+
+	err := v.Run()
+	if err != nil || v.HasFailed() || v.Estack().Len() != 1 {
 		return false
 	}
 
-	result, err := v.Estack().Top().TryBool()
+	res, err := v.Estack().Pop().TryBool()
```

Comment:
You can crash here with null pointer dereference.

Reply:
I have checked if `Estack().Len()` is equal to one earlier.

Reply:
OK, true.